### PR TITLE
Supress EmptyDSLErrors on skipped tests

### PR DESCRIPTION
**Pull request by brian-bates**
- Prevented `EmptyDSLErrors` from being raised on a `DataDrivenClass`
   that has been skipped.
 - Added metatests for the new functionality.
 - Fixed a random PEP8 error.
 - Fixed a 3.5 support issue.

### CODE REVIEW

**Comment by brian-bates** [comment]
Addresses issue https://github.com/CafeHub/opencafe/issues/48.

File: cafe/engine/models/data_interfaces.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ class ConfigParserDataSource(DataSource):
     def __init__(self, config_file_path, section_name):
         super(ConfigParserDataSource, self).__init__()
 
-        cafe_env_var = {key: value for key, value in os.environ.iteritems()
```
Comment (jidar) [inline cafe/engine/models/data_interfaces.py]: Please limit changes to PR scope.

File: cafe/plugins/http/cafe/engine/http/client.py [hunk 2]
```diff
@@ -105,7 +105,8 @@ def _wrapper(*args, **kwargs):
                 log.error("An exception occured durring logging")
                 log.exception()
 
-            response_header = '\n{0}\nRESPONSE RECEIVED\n{0}\n'.format('-' * 17)
+            response_header = '\n{0}\nRESPONSE RECEIVED\n{0}\n'.format(
```
Comment (jidar) [inline cafe/plugins/http/cafe/engine/http/client.py]: Same.

File: cafe/engine/models/data_interfaces.py (hunk 1 above)
Comment (jidar) [inline cafe/engine/models/data_interfaces.py]: If you want to change this to fix a python 3.X compatibility issue, please make that change in a separate PR.

Comment (brian-bates) [inline cafe/engine/models/data_interfaces.py]: I appreciate the feedback, but his change should be here; The increased coverage provided by new unittest associated with this PR uncover this bug. If we do not address it, the PR will fail gate jobs. Additionally, this follows the boy scout rule of leaving the code better than we found it.

File: cafe/plugins/http/cafe/engine/http/client.py (hunk 2 above)
Comment (brian-bates) [inline cafe/plugins/http/cafe/engine/http/client.py]: Because the gate jobs don't have PEP8 checks yet, I can separate this, but it seems silly to do so. I would expect we want every submission to be PEP8 compliant regardless of it not being gated.
